iceoryx_hoofs/source/posix_wrapper/thread.cpp

```diff
@@ -16,38 +16,97 @@
+            }
+        }
+        m_isJoinable = false;
+    }
 }
 
-ThreadName_t getThreadName(pthread_t thread) noexcept
+void Thread::setThreadName(const ThreadName_t& name) noexcept
+{
+    posixCall(iox_pthread_setname_np)(m_threadHandle, name.c_str())
+        .successReturnValue(0)
+        .evaluate()
+        .or_else([](auto& r) {
+            // String length limit is ensured through cxx::string
+            // ERANGE (string too long) intentionally not handled to avoid untestable and dead code
+            LogError() << "This should never happen! " << r.getHumanReadableErrnum();
+            cxx::Ensures(false && "internal logic error");
+            /// @todo thread specific comm file under /proc/self/task/[tid]/comm is read. Opening this file can fail
+            /// and errors possible for open(2) can be retrieved. Handle them here?
+        });
+}
+
+ThreadName_t Thread::getThreadName() noexcept
 {
     char tempName[MAX_THREAD_NAME_LENGTH + 1U];
 
-    posixCall(pthread_getname_np)(thread, tempName, MAX_THREAD_NAME_LENGTH + 1U)
+    posixCall(pthread_getname_np)(m_threadHandle, tempName, MAX_THREAD_NAME_LENGTH + 1U)
         .successReturnValue(0)
         .evaluate()
         .or_else([](auto& r) {
             // String length limit is ensured through MAX_THREAD_NAME_LENGTH
             // ERANGE (string too small) intentionally not handled to avoid untestable and dead code
-            std::cerr << "This should never happen! " << r.getHumanReadableErrnum() << std::endl;
+            LogError() << "This should never happen! " << r.getHumanReadableErrnum();
             cxx::Ensures(false && "internal logic error");
+            /// @todo thread specific comm file under /proc/self/task/[tid]/comm is read. Opening this file can fail
+            /// and errors possible for open(2) can be retrieved. Handle them here?
         });
 
     return ThreadName_t(cxx::TruncateToCapacity, tempName);
 }
 
+ThreadError Thread::errnoToEnum(const int errnoValue) noexcept
+{
+    switch (errnoValue)
+    {
+    case EAGAIN:
+        LogError() << "insufficient resources to create another thread";
+        return ThreadError::INSUFFICIENT_RESOURCES;
+    case EINVAL:
+        LogError() << "invalid attribute settings";
+        return ThreadError::INVALID_ATTRIBUTES;
+    case ENOMEM:
+        LogError() << "not enough memory to initialize the thread attributes object";
+        return ThreadError::INSUFFICIENT_MEMORY;
+    case EPERM:
+        LogError() << "no appropriate permission to set required scheduling policy or parameters";
+        return ThreadError::INSUFFICIENT_PERMISSIONS;
+    default:
+        LogError() << "an unexpected error occurred in thread - this should never happen! errno: "
+                   << strerror(errnoValue);
+        return ThreadError::UNDEFINED;
+    }
+}
+
+void* Thread::startRoutine(void* callable)
+{
+    // necessary because the callback signature for pthread_create is void*(void*)
```

Comment:
I would remove the comment. I think it is also out of date since you cast the `callable` argument to `void()` and not `void*(void*)`